CMakeLists.txt

```diff
@@ -87,6 +87,11 @@ if( "${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU|(Apple)?Clang" )
   endif()
 endif()
 
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+  # In Debug build, ensure that BSLS_ASSERT_SAFE macro is enabled
+  add_definitions("-DBSLS_ASSERT_LEVEL_ASSERT_SAFE")
```

Comment:
When built with `build-darwin.sh`, macro `BSLS_ASSERT_SAFE` is not enabled despite `Debug` build type set.

https://bloomberg.github.io/bde-resources/doxygen/bde_api_prod/group__bsls__assert.html

```We can build the our_square component in "safe mode" -- e.g., by incorporating -DBSLS_ASSERT_LEVEL_ASSERT_SAFE on the (Unix) command line.```